integration_tests/e2e/goal/archiveGoal.cy.ts

```diff
@@ -0,0 +1,142 @@
+    cy.task('stubGetFooterComponent')
+    cy.task('stubPrisonerList')
+    cy.task('stubCiagInductionList')
+    cy.task('stubActionPlansList')
+    cy.task('getPrisonerById')
+    cy.task('stubGetInduction')
+    cy.task('getActionPlan')
+    cy.task('stubLearnerProfile')
+    cy.task('stubLearnerEducation')
+    cy.task('archiveGoal')
+  })
+
+  it('should be able to navigate directly to archive goal page', () => {
+    // Given
+    const prisonNumber = 'G6115VJ'
+    const goalReference = '10efc562-be8f-4675-9283-9ede0c19dade'
+    cy.signIn()
+
+    // When
+    cy.visit(`/plan/${prisonNumber}/goals/${goalReference}/archive`)
+
+    // Then
+    const archiveGoalPage = Page.verifyOnPage(ArchiveGoalPage)
+    archiveGoalPage.isForGoal(goalReference)
+  })
+
+  it('should be able to navigate to archive goal page from overview', () => {
+    // Given
+    const prisonNumber = 'G6115VJ'
+    cy.signIn()
+
+    cy.visit(`/plan/${prisonNumber}/view/overview`)
+    const overviewPage = Page.verifyOnPage(OverviewPage)
+
+    // When
+    overviewPage.clickArchiveButtonForFirstGoal()
+
+    // Then
+    Page.verifyOnPage(ArchiveGoalPage)
+  })
+
+  it('should not submit the form if there are validation errors on the page', () => {
+    // Given
+    const prisonNumber = 'G6115VJ'
+    const goalReference = '10efc562-be8f-4675-9283-9ede0c19dade'
+    cy.signIn()
+    cy.visit(`/plan/${prisonNumber}/goals/${goalReference}/archive`)
+
+    // When
+    const archiveGoalPage = Page.verifyOnPage(ArchiveGoalPage)
+    archiveGoalPage.submitPage()
+
+    // Then
+    Page.verifyOnPage(ArchiveGoalPage)
+    archiveGoalPage //
+      .hasErrorCount(1)
+      .hasFieldInError('reason')
+  })
+
+  it('should not submit the form if other reason not entered', () => {
+    // Given
+    const prisonNumber = 'G6115VJ'
+    const goalReference = '10efc562-be8f-4675-9283-9ede0c19dade'
+    cy.signIn()
+    cy.visit(`/plan/${prisonNumber}/goals/${goalReference}/archive`)
+
+    // When
+    const archiveGoalPage = Page.verifyOnPage(ArchiveGoalPage)
+    archiveGoalPage //
+      .selectReason(ReasonToArchiveGoalValue.OTHER)
+      .submitPage()
+
+    // Then
+    Page.verifyOnPage(ArchiveGoalPage)
+    archiveGoalPage //
+      .hasErrorCount(1)
+      .hasFieldInError('reasonOther')
+  })
+
+  it('should ask for confirmation and return to overview if choosing no', () => {
+    // Given
+    const prisonNumber = 'G6115VJ'
+    const goalReference = '10efc562-be8f-4675-9283-9ede0c19dade'
+    cy.signIn()
+    cy.visit(`/plan/${prisonNumber}/goals/${goalReference}/archive`)
+    const archiveGoalPage = Page.verifyOnPage(ArchiveGoalPage)
+    archiveGoalPage //
+      .selectReason(ReasonToArchiveGoalValue.OTHER)
+      .enterReason('Just because...')
+      .submitPage()
+
+    // When
+    const reviewPage = Page.verifyOnPage(ReviewArchiveGoalPage)
+    reviewPage.clickNo()
+
+    // Then
+    Page.verifyOnPage(OverviewPage)
+  })
```

Comment:
Minor; perhaps we could also verify the API was NOT called in this case?
We have a cypress command that should do this, and I'm pretty sure we have a couple of cases where it's used
```
Cypress.Commands.add('wiremockVerifyNoInteractions', (requestPatternBuilder: RequestPatternBuilder) => {
  return cy.wrap(verify(0, requestPatternBuilder)).should('be.true')
})
```